### DISCUSSION TITLE
Is validation with JSR380 (or Spring validation) supported?

### DISCUSSION THREAD

**Discussion by frankkoornstra** [ExpediaGroup/graphql-kotlin#843] [Q&A] [answered]
I can't seem to find any examples or docs about this but it seems elementary so I'm probably missing something :sweat_smile: 

I have, for example, the following setup:
```kotlin
@Component
class MyAwesomeMutation : Mutation {
    fun myAwesomeMutation(widget: Widget) = widget
}

@Validated
data class Widget(@get:Min(3) val id: Int)
```
_(also tried it with `@field:Min(3)`)_

When I make a request for this mutation, it doesn't matter what I put in for the `id`, it will always be fine (as long as it's an `int` of course). Do I need to hook up something from Spring to validate the request?

I'm not tied to JSR380 perse, it just seemed like an obvious choice but if a Spring validation component works just as well, that's fine too. I just need validation to happen :smile:

**Comment by frankkoornstra** [comment, marked as answer]
Solved! Turned out I needed to shift the annotations around a bit. This is the working solution:
```kotlin
@Component
@Validated
class MyAwesomeMutation : Mutation {
    fun myAwesomeMutation(@Valid widget: Widget) = widget
}

data class Widget(@get:Min(3) val id: Int)
```

**Comment by dariuszkuc** [comment]
FYI if you are using SpringBoot 2.3.x+ web starters no longer bring in the validation starter as a dependency - you will have to add it explicitly, e.g. in Gradle add following dependency

```
implementation("org.springframework.boot", "spring-boot-starter-validation", springBootVersion)
```
